Jenkinsfile

```diff
@@ -29,225 +23,154 @@ pipeline {
+                    colourText("info","Stage: ${env.STAGE} successful!")
                 }
                 failure {
-                    colourText("warn","Failed on Tests.")
+                    colourText("warn","Stage: ${env.STAGE} failed!")
                 }
             }
         }
 
-
-        // bundle all libs and dependencies
-        stage ('Bundle') {
+        stage('Static Analysis') {
             agent any
-            when {
-                anyOf {
-                    branch BRANCH_DEV
-                    branch BRANCH_TEST
-                    branch BRANCH_PROD
-                }
-            }
+            environment{ STAGE = "Static Analysis" }
             steps {
-                script {
-                    env.NODE_STAGE = "Bundle"
+                parallel (
+                    "Scalastyle" : {
+                        colourText("info","Running scalastyle analysis")
+                        sh "sbt scalastyle"
+                    },
+                    "Scapegoat" : {
+                        colourText("info","Running scapegoat analysis")
+                        sh "sbt scapegoat"
+                    }
+                )
+            }
+            post {
+                success {
+                    step([$class: 'CheckStylePublisher', pattern: '**/target/code-quality/style/*scalastyle*.xml, **/target/*/scapegoat-report/scapegoat-scalastyle.xml'])
+                    colourText("info","Stage: ${env.STAGE} successful!")
                 }
-                colourText("info", "Bundling....")
-                dir('conf') {
-                    git(url: "$GITLAB_URL/BusinessIndex/${MODULE_NAME}.git", credentialsId: GITLAB_CREDS, branch: "develop")
+                failure {
+                    colourText("warn","Stage: ${env.STAGE} failed!")
                 }
-                // stash name: "zip"
             }
         }
 
-        stage("Releases"){
+        stage('Package'){
             agent any
-            when {
-                anyOf {
-                    branch BRANCH_DEV
-                    branch BRANCH_TEST
-                    branch BRANCH_PROD
-                }
+            when{ expression{ isBranch("master") }}
+            environment{ 
+                STAGE = "Package" 
+                DEPLOY_TO = "dev"    
             }
             steps {
-                script {
-                    env.NODE_STAGE = "Releases"
-                    currentTag = getLatestGitTag()
-                    colourText("info", "Found latest tag: ${currentTag}")
-                    newTag =  IncrementTag( currentTag, RELEASE_TYPE )
-                    colourText("info", "Generated new tag: ${newTag}")
-                    //push(newTag, currentTag)
+                dir('gitlab') {
+                    git(url: "$GITLAB_URL/BusinessIndex/${MODULE_NAME}.git", credentialsId: GITLAB_CREDS, branch: "master")
                 }
```

Comment:
The git clone isnt required yet. This can be moved to deploy.

Reply:
Moved